apps/condo/domains/user/schema/ConfirmEmailActionService.js

```diff
@@ -314,15 +323,47 @@ const ConfirmEmailActionService = new GQLCustomSchema('ConfirmEmailActionService
                 const isInvalidData = await redisGuard.isLocked(captcha, 'validation-failed')
                 if (isInvalidData) return { token }
 
-                await ConfirmEmailAction.create(context, payload)
+                const confirmAction = await ConfirmEmailAction.create(context, payload, 'id')
 
-                await sendMessage(context,  {
-                    to: { email: normalizedEmail },
-                    type: EMAIL_VERIFY_CODE_MESSAGE_TYPE,
-                    meta: {
+                let meta = {}
+
+                if (messageType === VERIFY_USER_EMAIL_MESSAGE_TYPE) {
+                    const tokenPayload = {
+                        messageType,
+                        secretCode,
+                        token,
+                    }
+                    const tokenInString = JSON.stringify(tokenPayload)
+                    const encodedToken = encryptionManager.encrypt(tokenInString)
+                    const encodedTokenInBase64 = base64UrlEncode(encodedToken)
+                    const linkRaw = new URL(SERVER_URL)
+                    linkRaw.pathname = '/user/confirmEmail'
+                    linkRaw.searchParams.set('token', encodedTokenInBase64)
+                    const link = linkRaw.href
+
+                    meta = {
+                        dv: 1,
+                        link,
+                        withMarketingConsent: VERIFY_USER_EMAIL_WITH_MARKETING_CONSENT_ENABLED,
+                    }
+
+                    // NOTE: The user may open the link from the email later than one minute later
+                    await ConfirmEmailAction.update(context, confirmAction.id, {
+                        dv: 1,
+                        sender,
+                        secretCodeExpiresAt: expiresAt,
```

Comment:
change name of `expiresAt` const to more specific, mb `confirmActionExpiresAt`

Reply:
Done